### PR TITLE
Update resource manager to share resources between parallel tests

### PR DESCRIPTION
**Pull request by kornys**
### Type of change

- Enhancement / new feature

### Description

Allow resource manager to share default resources created by annotations between parallel tests

### Checklist

- [ ] Update/write design documentation in `./documentation/design`
- [ ] Write tests and make sure they pass
- [ ] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md

### CODE REVIEW

**Comment by kornys** [comment]
@enmasse-ci run tests test_only crc

**Comment by kornys** [comment]
@enmasse-ci run tests test_only crc

**Comment by kornys** [comment]
@enmasse-ci run tests test_only crc

**Comment by kornys** [comment]
@enmasse-ci run tests test_only crc

**Comment by kornys** [comment]
@enmasse-ci run tests test_only crc

**Comment by kornys** [comment]
@enmasse-ci run tests test_only crc

**Comment by kornys** [comment]
@enmasse-ci run tests test_only crc

File: systemtests/src/main/java/io/enmasse/systemtest/messaginginfra/resources/ResourceKind.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+/*
+ * Copyright 2020, EnMasse authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.enmasse.systemtest.messaginginfra.resources;
+
+public interface ResourceKind {
```
Comment (obabec) [inline systemtests/src/main/java/io/enmasse/systemtest/messaginginfra/resources/ResourceKind.java:7]: I would consider if it wouldn't be better to use enom as long as we don't implement this interface and only using it as "static" call.

**Pull request merged**